### PR TITLE
Provide the guide for creating diagrams with draw.io

### PR DESCRIPTION
**Pull request by vpantasenko**
Provide the basic guide for creating diagrams with draw.io web application.

### CODE REVIEW

File: SUMMARY.md [hunk 1]
```diff
@@ -5,6 +5,7 @@
   * [Project Overview](project-documentation/project-overview.md)
   * Product Requirements
   * Diagrams
+  * [Diagrams Guide](project-documentation/diagrams-guide.md)
```
Comment (aleksey-pastuhov) [inline SUMMARY.md]: Please move it under `Development -> Common processes`

File: project-documentation/diagrams-guide.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
+# Working with diagrams
+
+For creating a diagram it's recommended to use [draw.io](https://www.draw.io) web application. The current guide describes the process of creating diagrams in draw.io application.
```
Comment (aleksey-pastuhov) [inline project-documentation/diagrams-guide.md]: the process of creating and editing diagrams.

File: project-documentation/diagrams-guide.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+## Creating a diagram
+
+1. Create new feature branch under your repository (for example `feature/provide-use-case-diagram`).
+2. Make sure that directory structure for the future diagram is prepared (for example new `Use Case Diagram.xml` file will be in the `docs/diagrams` directory).
```
Comment (aleksey-pastuhov) [inline project-documentation/diagrams-guide.md]: the `diagrams` directory)

File: project-documentation/diagrams-guide.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+3. Open [draw.io](https://www.draw.io) web application in your web browser.
+4. In the *Save diagrams to* pop-up click *Decide later* button in the bottom. We'll configure saving in further steps.
```
Comment (aleksey-pastuhov) [inline project-documentation/diagrams-guide.md]: ?? I do not have that pop-up

File: project-documentation/diagrams-guide.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+10. Choose *GitHub* option.
+11. In the *Authorization required* pop-up press *Authorise* to authorise draw.io application using your GitHub account.
+12. Enter your GitHub credentials in the new opened window and press *sign in*.
+13. In the *Select Folder* pop-up you'll see a list of repositories available to save new diagram.
+14. After selecting the repository you'll be asked to choose folder under it.
+15. Choose the branch created in the first step (in our example it is `feature/provide-use-case-diagram`) by clicking on master branch (you should see something like this `<username>/<repository_name> / master`).
+16. In the opened branch list choose your feature branch.
+17. Select directory to save the diagram and press *OK* button.
+18. In the opened pop-up enter commit message and press *OK* button.
+19. Go to your repository to make sure that file *Use Case Diagram.xml* file is created in the `docs/diagrams` directory under `feature/provide-use-case-diagram` branch.
```
Comment (aleksey-pastuhov) [inline project-documentation/diagrams-guide.md]: created in the `dev-docs` repository in the `diagrams` directory

File: project-documentation/diagrams-guide.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+20. To get a link to created diagram go to *File > Publis > Link...*, in the opened pop-up leave default options and press *Create*. In the next opened pop-up link to current diagram will be provided.
```
Comment (aleksey-pastuhov) [inline project-documentation/diagrams-guide.md]: Publish

File: project-documentation/diagrams-guide.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+
+## Editing a diagram
+
+1. Open [draw.io](https://www.draw.io) web application in your web browser.
+2. In the *GitHub* pop-up choose *Open Existing Diagram*.
+3. In the *Select Folder* pop-up chose repository in which you want to find a diagram file.
+4. Choose branch you are working on (`master` branch is used by default).
+5. Find and open diagram file in the directory structure.
+6. Now your diagram is loaded and you are able to continue to work with it.
+
```
Comment (aleksey-pastuhov) [inline project-documentation/diagrams-guide.md]: Please add one point that those who edit need to describe changes on the diagram in PR and add links to previous and new versions of diagrams to compare.

File: SUMMARY.md (hunk 1 above)
Comment (vpantasenko) [inline SUMMARY.md]: +++

File: project-documentation/diagrams-guide.md (hunk 2 above)
Comment (vpantasenko) [inline project-documentation/diagrams-guide.md]: +++

File: project-documentation/diagrams-guide.md (hunk 3 above)
Comment (vpantasenko) [inline project-documentation/diagrams-guide.md]: +++

File: project-documentation/diagrams-guide.md (hunk 4 above)
Comment (vpantasenko) [inline project-documentation/diagrams-guide.md]: Updated steps to fit cases when a user is authorised in the draw.io and when they aren't.

File: project-documentation/diagrams-guide.md (hunk 5 above)
Comment (vpantasenko) [inline project-documentation/diagrams-guide.md]: +++

File: project-documentation/diagrams-guide.md (hunk 6 above)
Comment (vpantasenko) [inline project-documentation/diagrams-guide.md]: Changed to `File > Export as > URL...`. It saves a snapshot of the current drawing. The previous case saved only last snapshot.

File: project-documentation/diagrams-guide.md (hunk 7 above)
Comment (vpantasenko) [inline project-documentation/diagrams-guide.md]: +++

**Comment by vpantasenko** [comment]
[Common processes](https://trello.com/c/F5HJuEjT/18-common-processes)

**Pull request merged**